tests/sort_test.py

```diff
@@ -75,3 +74,10 @@ def testErrorHandling(self):
             # Test attempt to sort Strings object, which is unsupported
             with self.assertRaises(TypeError):
                 ak.sort(ak.array(["String {}".format(i) for i in range(0, 10)]), algo)
+
+    def test_nan_sort(self):
+        # Reproducer from #2703
+        np_arr = np.array([-3.14, np.inf, np.nan, -np.inf, 3.14, 0.0, 3.14, -8])
```

Comment:
I think it'd be worth having a positive version with nan/inf as well.

Reply:
a version with only positive values? That's a good idea, just to make sure it still handles things correctly when `negs` is False